### PR TITLE
pybind: Misc fixes to ceph_volume_client.py

### PR DESCRIPTION
**Pull request by onyb**
Continuation of a1347086f82a34e609268cb09e9a9b9c1f68aafe

Signed-off-by: Anirudha Bose ani07nov@gmail.com

### CODE REVIEW

**Comment by jdurgin** [comment]
lgtm

**Comment by tchaikov** [comment]
lgtm also.

**Comment by ajarr** [comment]
@gregsfortytwo, please do not merge this. I'll post my review comments tomorrow.

File: src/pybind/ceph_volume_client.py [hunk 1]
```diff
@@ -318,7 +317,7 @@ def recover(self):
 
         log.debug("Recovered from partial auth updates (if any).")
 
-    def _recover_auth_meta(auth_id, auth_meta):
+    def _recover_auth_meta(self, auth_id, auth_meta):
```
Comment (ajarr) [inline src/pybind/ceph_volume_client.py]: Can we exclude this fix from the commit? Rest of the fixes in the commit seem to be coding style or py3 compat related. I think it'd be better to file a separate tracker for the broken recover phase. I also spotted other bugs with this part of the code. I can file the tracker and fix all the issues related to the recovery as a different PR.

**Comment by ajarr** [comment]
Unlike indicated in the commit message, most of the fixes do not seem to be py3 compat related. The only py3 compat fix I can spot is the conversion of xattr value to byte type when setting xattr. If I'm right, then maybe it would be better to split the commit into ones that fix py3 compat and the ones that don't?

File: src/pybind/ceph_volume_client.py [hunk 2]
```diff
@@ -617,7 +616,7 @@ def create_volume(self, volume_path, size=None, data_isolated=False):
         self._mkdir_p(path)
 
         if size is not None:
-            self.fs.setxattr(path, 'ceph.quota.max_bytes', size.__str__(), 0)
+            self.fs.setxattr(path, 'ceph.quota.max_bytes', str(size).encode(), 0)
```
Comment (ajarr) [inline src/pybind/ceph_volume_client.py]: Lines 631 and 636 also need similar fixes, right?

**Comment by gregsfortytwo** [comment]
This passed testing but I'm leaving it to discuss with @ajarr and @onyb now. :)

File: src/pybind/ceph_volume_client.py (hunk 1 above)
Comment (ajarr) [inline src/pybind/ceph_volume_client.py]: @onyb I've created a bug tracker for this issue that I'd like work to on. Please let me know if you're OK with me taking this up and other broken recovery related problems?
http://tracker.ceph.com/issues/17216

**Comment by ajarr** [comment]
@onyb, ping.

I've created a feature tracker (http://tracker.ceph.com/issues/17230) for py3 compat of ceph_volume_client. It should help track this work and also mentions what needs to be done. It'd be nice to add commit message tag ,
Fixes: http://tracker.ceph.com/issues/17230, to the ceph_volume_client's py3 related fixes.

File: src/pybind/ceph_volume_client.py (hunk 2 above)
Comment (onyb) [inline src/pybind/ceph_volume_client.py]: @ajarr Correct. Adding a simple `.encode()` should suffice.

**Comment by onyb** [comment]
@ajarr Since the changes were not huge, I clubbed everything in one commit and named it "Misc fixes ..". On your suggestion, I'm removing the changes which are not related to py3 compat, and also fixing the two other calls to `setxattr`. Stay tuned. :)

**Comment by jcsp** [comment]
@ajarr are you happy with this now?  I don't mind having the _recover_auth_meta fix in this PR if there isn't another PR outstanding that would also fix it

**Comment by jcsp** [comment]
(or @onyb are you still working on this?)

**Comment by ajarr** [comment]
@jcsp, I thought @onyb agreed to upload a new patch set based on my comments from a while ago. I don't see that patch set yet.

I'm working on the PR to fix recovery of auth meta, http://tracker.ceph.com/issues/17230. Isn't it better if a PR is related to a tracker issue , so that it's easier to backport (to Jewel)? I'll upload the PR soon. Sorry about the delay.

**Comment by jcsp** [comment]
@onyb please rebase this on master

**Comment by jcsp** [comment]
@onyb are you coming back to this or shall I close it?

**Comment by onyb** [comment]
@jcsp Please close this, as I don't have time at the moment to split the PR. @ajarr Kindly borrow parts of the patch, if still relevant.

**Pull request closed without merging**